src/utils/eip712/parser.rs

```diff
@@ -127,18 +119,52 @@ mod tests {
     #[test]
     fn test_parser() {
         let source = "byte[][][7][][][][][][][]";
```

Comment:
This looks wrong, there's no atomic type called `byte`, though there is `bytes1`. To me it looks like this test should not pass. See: https://eips.ethereum.org/EIPS/eip-712#definition-of-typed-structured-data-%F0%9D%95%8A

Reply:
This is actually correct, because `byte` is valid custom type identifier.